src/core/jsproxy.c

```diff
@@ -1500,6 +1503,174 @@ static PyMethodDef JsProxy_finally_MethodDef = {
   METH_O,
 };
 
+// Forward declaration
+static PyObject*
+create_proxy_of_type(int type_flags, JsRef object, JsRef this);
+
+static PyObject*
+JsProxy_as_object_map(PyObject* self, PyObject* Py_UNUSED(ignored))
+{
+  int type_flags = IS_OBJECT_MAP;
+  return create_proxy_of_type(
+    type_flags, JsProxy_REF(self), JsMethod_THIS(self));
+}
+
+static PyMethodDef JsProxy_as_object_map_MethodDef = {
+  "as_object_map",
+  (PyCFunction)JsProxy_as_object_map,
+  METH_NOARGS,
+};
+
+EM_JS_REF(JsRef, JsObjMap_GetIter_js, (JsRef idobj), {
+  let jsobj = Hiwire.get_value(idobj);
+  return Hiwire.new_value(Module.iterObject(jsobj));
+})
+
+static PyObject*
+JsObjMap_GetIter(PyObject* self)
+{
+  JsRef iditer = JsObjMap_GetIter_js(JsProxy_REF(self));
+  if (iditer == NULL) {
+    return NULL;
+  }
+  PyObject* result = js2python(iditer);
+  hiwire_decref(iditer);
+  return result;
+}
+
+EM_JS_NUM(int, JsObjMap_length_js, (JsRef idobj), {
+  let jsobj = Hiwire.get_value(idobj);
+  let length = 0;
+  for (let _ of Module.iterObject(jsobj)) {
+    length++;
+  }
+  return length;
```

Comment:
Maybe we need to mention somewhere that it takes `O(n)` for `len(jsproxy_object_map)`?

Reply:
Yeah perhaps this is worth pointing out. This is a good reason to use a `Map` instead =)